src/Library/demos/Drop Down/main.js

```diff
@@ -0,0 +1,68 @@
+import Gtk from "gi://Gtk";
+import GObject from "gi://GObject";
+import Gio from "gi://Gio";
+
+const drop_down = workbench.builder.get_object("drop_down");
+const advanced_drop_down = workbench.builder.get_object("advanced_drop_down");
+
+drop_down.connect("notify::selected-item", () => {
+  const selected_item = drop_down.selected_item.get_string();
+  console.log(selected_item);
+});
+
+const KeyValuePair = GObject.registerClass(
+  {
+    Properties: {
+      key: GObject.ParamSpec.string(
+        "key",
+        null,
+        null,
+        GObject.ParamFlags.READWRITE,
+        "",
+      ),
+      value: GObject.ParamSpec.string(
+        "value",
+        "Value",
+        "Value",
+        GObject.ParamFlags.READWRITE,
+        "",
+      ),
+    },
+  },
+  class KeyValuePair extends GObject.Object {},
+);
+
+const model = new Gio.ListStore();
+
+model.append(new key_value_pair({ key: "win7", value: "Windows 7" }));
+model.append(new key_value_pair({ key: "win10", value: "Windows 10" }));
+model.append(new key_value_pair({ key: "win11", value: "Windows 11" }));
+model.append(new key_value_pair({ key: "ubuntu", value: "Ubuntu" }));
+model.append(new key_value_pair({ key: "fedora", value: "Fedora" }));
+model.append(new key_value_pair({ key: "debian", value: "Debian" }));
+model.append(new key_value_pair({ key: "mint", value: "Mint" }));
+model.append(new key_value_pair({ key: "arch", value: "Arch Linux" }));
+model.append(new key_value_pair({ key: "popos", value: "Pop!_OS" }));
+model.append(new key_value_pair({ key: "opensuse", value: "OpenSUSE" }));
+model.append(new key_value_pair({ key: "gentoo", value: "Gentoo" }));
+model.append(new key_value_pair({ key: "freebsd", value: "FreeBSD" }));
+model.append(new key_value_pair({ key: "macos", value: "macOS" }));
+model.append(new key_value_pair({ key: "ios", value: "iOS" }));
+model.append(new key_value_pair({ key: "android", value: "Android" }));
```

Comment:
```suggestion
model.splice(0, 0, [
  new KeyValuePair({ key: "win7", value: "Windows 7" }),
  new KeyValuePair({ key: "win10", value: "Windows 10" }),
  new KeyValuePair({ key: "win11", value: "Windows 11" }),
  new KeyValuePair({ key: "ubuntu", value: "Ubuntu" }),
  new KeyValuePair({ key: "fedora", value: "Fedora" }),
  new KeyValuePair({ key: "debian", value: "Debian" }),
  new KeyValuePair({ key: "mint", value: "Mint" }),
  new KeyValuePair({ key: "arch", value: "Arch Linux" }),
  new KeyValuePair({ key: "popos", value: "Pop!_OS" }),
  new KeyValuePair({ key: "opensuse", value: "OpenSUSE" }),
  new KeyValuePair({ key: "gentoo", value: "Gentoo" }),
  new KeyValuePair({ key: "freebsd", value: "FreeBSD" }),
  new KeyValuePair({ key: "macos", value: "macOS" }),
  new KeyValuePair({ key: "ios", value: "iOS" }),
  new KeyValuePair({ key: "android", value: "Android" }),
]);
```

The main advantage here, other than cleaner code, is you only get one emission of `items-changed`, instead of one per item.

I'll let you adjust the subject matter ;)